File: python/bundled_agents/bridge/bridge_core/src/bridge_core/ingress_rules/allow_all.py
"""`allow_all` — the default ingress rule: full symmetric duplex, a true no-op."""

from __future__ import annotations

from dataclasses import dataclass

from bridge_core._authorizer import ALLOW, Action, Decision, IngressRule


@dataclass(frozen=True)
class AllowAll(IngressRule):
    """Permit every inbound action. The engine default (absent rule ⇒ this)."""

    def authorize(self, action: Action) -> Decision:
        return ALLOW

File: rust/crates/bundles/fantastic-kernel-bridge/src/authorizer/ingress/allow_all.rs
//! `allow_all` — the default ingress rule: full symmetric duplex, a true no-op.

use super::super::{Action, Decision, IngressRule};

/// Permit every inbound action. The engine default (absent rule ⇒ this).
pub struct AllowAll;

impl IngressRule for AllowAll {
    fn authorize(&self, _action: &Action) -> Decision {
        Decision::Allow
    }
}
